lib/watir/locators/element/locator.rb

```diff
@@ -165,10 +171,10 @@ def create_filter_selector
           @filter_selector
         end
 
-        def process_label
-          return unless @normalized_selector[:label].kind_of?(Regexp) && selector_builder.should_use_label_element?
+        def process_label(label_key)
+          return unless @normalized_selector[label_key].kind_of?(Regexp) && selector_builder.should_use_label_element?
```

Comment:
This check will need to be different based on the label key:
* :label is only processed if a Regexp (no change)
* :visible_label is always processed - ie regardless of whether String or Regexp

Right now, `browser.text_field(visible_label: 'text')` will go into the XPath builder and be treated like a custom attribute.

Reply:
ugh, you're right. sigh.